src/callbacks_step/euler_acoustics_coupling.jl

```diff
@@ -0,0 +1,190 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+"""
+    EulerAcousticsCouplingCallback
+
+!!! warning "Experimental code"
+    This callback is experimental and may change in any future release.
+
+A callback that couples the acoustic perturbation equations and compressible Euler equations. Must
+be used in conjunction with [`SemidiscretizationEulerAcoustics`](@ref).
+This callback manages the flow solver - which is always one time step ahead of the
+acoustics solver - and calculates the acoustic source terms based on the linearized lamb vector of
+the flow solution after each time step. Furthermore, it manages the step size for both solvers
+and initializes the mean values of the acoustic perturbation equations using results obtained with
+the [`AveragingCallback`](@ref).
+"""
```

Comment:
I guess this approach is based on some paper? It would be nice to include this as reference

Reply:
Added a reference in 8a7a3aa.